corehq/apps/locations/util.py

```diff
@@ -129,18 +129,32 @@ def __init__(self, domain, include_consumption=False, root_location_id=None,
         self._location_count = None
         self._locations_exported = 0
 
+        additional_filters = self._extract_additional_filters(**kwargs)
+        selected_location_only = kwargs.get('selected_location_only', False)
+
         if headers_only:
             self.base_query = SQLLocation.objects.none()
         elif root_location_id:
-            root_location = SQLLocation.objects.get(location_id=root_location_id)
-            self.base_query = SQLLocation.active_objects.get_descendants(
-                Q(domain=self.domain, id=root_location.id)
-            )
+            if selected_location_only:
+                # Use filter so base_query is a LocationQuerySet
+                self.base_query = SQLLocation.objects.filter(location_id=root_location_id)
+            else:
+                root_location = SQLLocation.objects.get(location_id=root_location_id)
+                self.base_query = SQLLocation.objects.get_descendants(
+                    Q(domain=self.domain, id=root_location.id, **additional_filters)
+                )
         else:
-            self.base_query = SQLLocation.active_objects.filter(
+            self.base_query = SQLLocation.objects.filter(
```

Comment:
@kaapstorm Good question! 
Previously only active locations were exported, but now because the web user will have the option on the UI to export "All", "Only active" or "Only archived" locations, I changed this logic a bit.
For "All" selected on UI (this is also the default), we need to select all locations in a certain domain _without_ setting the `is_archived` filter, because setting it to `False` excludes archived locations and setting it to `True` excludes active locations. Only by not including the `is_archived` key in the filters will it select both the active and archived locations.